FEM: Fix parse diff function

lyphrowny opened the pull request:
[The function](https://github.com/FreeCAD/FreeCAD/blob/1ccb74f141ec4d1276b39d18fb99f276d09e007e/src/Mod/Fem/femtest/app/support_utils.py#L236) responsible for parsing the unified diff of the original .inp file and the one obtained by running the test worked incorrectly. With this PR this function:

- Correctly handles block changes
- Doesn't ignore lines consisting of single characters (added \ removed newline)

The tests for this function have been added to the "TestFemApp" suite. The tests are now under LGPL2+ license (it was missing)

Also added commented out ccxtools tests as per this comment: https://github.com/FreeCAD/FreeCAD/pull/13292#issuecomment-2037431281

lyphrowny committed:
FEM: fix logic in parse_diff

lyphrowny committed:
FEM: fix tests for parse_diff function

lyphrowny committed:
FEM: add tests for parse_diff in FemApp test suite

lyphrowny committed:
FEM: add license notice

lyphrowny committed:
FEM: add commented out tests for ccxtools

Were commented out because of comparing floats as strings (see comment: https://github.com/FreeCAD/FreeCAD/pull/13292#issuecomment-2037431281)

lyphrowny committed:
FEM: fix rewritten license notices; fix typos in comments

chennes commented:
Thanks for the PR -- can you please resolve the conflicts that got generated in the interim?

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

lyphrowny committed:
FEM: fix logic bug in _needs_further_action function

The .split() method was missing
The indexing of the split line was wrong (+1 to the needed index)
The semantic of the _needs_further_action was wrong in case of a multiline hunk block: further investigation was needed when the number of lines was not equal, which is wrong. Only when the number of lines is equal the hunk should be further investigated

lyphrowny committed:
FEM: remove extra newline in test's .inp files

lyphrowny committed:
FEM: improve functions' readability

lyphrowny commented:
* Resolved conflicts by rebasing onto main
* Fixed a semantic bug:
Forgot to call a `.split()` method on a string, which caused erroreus execution flow
* Did my best to improve readability of the functions and their names
  
  - Some function names were confusing (`_needs_further_action` - what action is needed? if the action is not needed, then what?)
  - Simplified logical expressions
  - Improved docstrings - they should be more clear now

lyphrowny commented:
@marioalexis84 Could you elaborate on that? What errors do you get?

The CI tests pass for both the current version of the branch and for an up-to-date one (merged "main" into it) ([results of the CI tests for the up-to-date branch](https://github.com/lyphrowny/FreeCAD/actions/runs/9931164587))

The files you mention are changed in this branch - might you be running the test suites with these files unchanged?

chennes commented:
Holding off on merging until we hear back from @marioalexis84

marioalexis84 commented:
@lyphrowny I still get fails on those two tests:

test_constraint_centrif:
```
Traceback (most recent call last):
  File "/usr/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "FreeCAD-test/Mod/Fem/femtest/app/test_ccxtools.py", line 226, in test_constraint_centrif
    self.input_file_writing_test(get_namefromdef("test_"))
  File "FreeCAD-test/Mod/Fem/femtest/app/test_ccxtools.py", line 372, in input_file_writing_test
    self.assertFalse(ret, f"ccxtools write_inp_file test failed.\n{ret}")
AssertionError: 'Comparing FreeCAD-test/Mod/Fem/femtest/data/calculix/constraint_centrif.inp to /tmp/FEM_unittests/ccxtools_constraint_centrif_8ffda9a3155f/Mesh.inp failed!\n@@ -18683,769 +18683,40 @@\n-1,\n-2,\n-10,\n-11,\n-12,\n-13,\n-14,\n-29,\n-30,\n-31,\n-32,\n-33,\n-34,\n-35,\n-36,\n-37,\n-38,\n-39,\n-40,\n-41,\n-42,\n-43,\n-44,\n-45,\n-46,\n-47,\n-48,\n-49,\n-50,\n-51,\n-52,\n-53,\n-54,\n-55,\n-56,\n-57,\n-58,\n-59,\n-60,\n-61,\n-62,\n-63,\n-64,\n-65,\n-66,\n-67,\n-68,\n-69,\n-70,\n-71,\n-72,\n-73,\n-74,\n-75,\n-76,\n-77,\n-78,\n-79,\n-80,\n-81,\n-82,\n-83,\n-84,\n-85,\n-86,\n-87,\n-88,\n-89,\n-90,\n-91,\n-92,\n-93,\n-94,\n-95,\n-96,\n-97,\n-98,\n-99,\n-100,\n-101,\n-102,\n-103,\n-104,\n-105,\n-106,\n-107,\n-108,\n-109,\n-110,\n-111,\n-112,\n-113,\n-114,\n-115,\n-116,\n-117,\n-118,\n-119,\n-120,\n-121,\n-122,\n-123,\n-124,\n-125,\n-126,\n-127,\n-128,\n-129,\n-130,\n-131,\n-132,\n-133,\n-134,\n-135,\n-136,\n-137,\n-138,\n-139,\n-140,\n-141,\n-142,\n-143,\n-144,\n-145,\n-146,\n-147,\n-148,\n-149,\n-150,\n-151,\n-152,\n-153,\n-154,\n-155,\n-156,\n-157,\n-158,\n-159,\n-160,\n-161,\n-162,\n-163,\n-164,\n-165,\n-166,\n-167,\n-168,\n-169,\n-170,\n-171,\n-172,\n-173,\n-174,\n-175,\n-176,\n-177,\n-178,\n-179,\n-180,\n-181,\n-182,\n-183,\n-184,\n-185,\n-186,\n-187,\n-188,\n-189,\n-190,\n-191,\n-192,\n-193,\n-194,\n-195,\n-196,\n-197,\n-198,\n-199,\n-200,\n-201,\n-202,\n-203,\n-204,\n-205,\n-206,\n-207,\n-208,\n-209,\n-210,\n-211,\n-212,\n-213,\n-214,\n-215,\n-216,\n-217,\n-218,\n-219,\n-220,\n-221,\n-222,\n-223,\n-224,\n-225,\n-226,\n-227,\n-228,\n-229,\n-230,\n-231,\n-232,\n-233,\n-234,\n-235,\n-236,\n-237,\n-238,\n-239,\n-240,\n-241,\n-242,\n-243,\n-244,\n-245,\n-246,\n-247,\n-248,\n-249,\n-250,\n-251,\n-252,\n-253,\n-254,\n-255,\n-256,\n-257,\n-258,\n-259,\n-260,\n-261,\n-262,\n-263,\n-264,\n-265,\n-266,\n-267,\n-268,\n-269,\n-270,\n-271,\n-272,\n-273,\n-274,\n-275,\n-276,\n-277,\n-278,\n-279,\n-280,\n-281,\n-282,\n-283,\n-284,\n-285,\n-523,\n-524,\n-525,\n-526,\n-527,\n-528,\n-529,\n-530,\n-531,\n-532,\n-533,\n-534,\n-535,\n-536,\n-537,\n-538,\n-539,\n-540,\n-541,\n-542,\n-543,\n-544,\n-545,\n-546,\n-547,\n-548,\n-549,\n-550,\n-551,\n-552,\n-553,\n-554,\n-555,\n-556,\n-557,\n-558,\n-559,\n-560,\n-561,\n-562,\n-563,\n-564,\n-565,\n-566,\n-567,\n-568,\n-569,\n-570,\n-571,\n-572,\n-573,\n-574,\n-575,\n-576,\n-577,\n-578,\n-579,\n-580,\n-581,\n-582,\n-583,\n-584,\n-585,\n-586,\n-587,\n-588,\n-589,\n-590,\n-591,\n-592,\n-593,\n-594,\n-595,\n-596,\n-597,\n-598,\n-599,\n-600,\n-601,\n-602,\n-603,\n-604,\n-605,\n-606,\n-607,\n-608,\n-609,\n-610,\n-649,\n-650,\n-651,\n-652,\n-653,\n-654,\n-655,\n-656,\n-657,\n-658,\n-659,\n-660,\n-661,\n-662,\n-663,\n-664,\n-665,\n-666,\n-667,\n-668,\n-669,\n-670,\n-671,\n-672,\n-673,\n-674,\n-675,\n-676,\n-677,\n-1337,\n-1338,\n-1339,\n-1340,\n-1341,\n-1342,\n-1343,\n-1344,\n-1345,\n-1346,\n-1347,\n-1348,\n-1349,\n-1350,\n-1351,\n-1352,\n-1353,\n-1354,\n-1355,\n-1356,\n-1357,\n-1358,\n-1359,\n-1360,\n-1361,\n-1362,\n-1363,\n-1364,\n-1365,\n-1366,\n-1367,\n-1368,\n-1369,\n-1370,\n-1371,\n-1372,\n-1373,\n-1374,\n-1375,\n-1376,\n-1377,\n-1378,\n-1379,\n-1380,\n-1381,\n-1382,\n-1383,\n-1384,\n-1385,\n-1386,\n-1387,\n-1388,\n-1389,\n-1390,\n-1391,\n-1392,\n-1393,\n-1394,\n-1395,\n-1396,\n-1397,\n-1398,\n-1399,\n-1400,\n-1401,\n-1402,\n-1403,\n-1404,\n-1405,\n-1406,\n-1407,\n-1408,\n-1409,\n-1410,\n-1411,\n-1412,\n-1413,\n-1414,\n-1415,\n-1416,\n-1417,\n-1418,\n-1419,\n-1420,\n-1421,\n-1422,\n-1423,\n-1424,\n-1425,\n-1426,\n-1427,\n-1428,\n-1429,\n-1430,\n-1431,\n-1432,\n-1433,\n-1434,\n-1435,\n-1436,\n-1437,\n-1438,\n-1439,\n-1440,\n-1441,\n-1442,\n-1443,\n-1444,\n-1445,\n-1446,\n-1447,\n-1448,\n-1449,\n-1450,\n-1451,\n-1452,\n-1453,\n-1454,\n-1455,\n-1456,\n-1457,\n-1458,\n-1459,\n-1460,\n-1461,\n-1462,\n-1463,\n-1464,\n-1465,\n-1466,\n-1467,\n-1468,\n-1469,\n-1470,\n-1471,\n-1472,\n-1473,\n-1474,\n-1475,\n-1476,\n-1477,\n-1478,\n-1479,\n-1480,\n-1481,\n-1482,\n-1483,\n-1484,\n-1485,\n-1486,\n-1487,\n-1488,\n-1489,\n-1490,\n-1491,\n-1492,\n-1493,\n-1494,\n-1495,\n-1496,\n-1497,\n-1498,\n-1499,\n-1500,\n-1501,\n-1502,\n-1503,\n-1504,\n-1505,\n-1506,\n-1507,\n-1508,\n-1509,\n-1510,\n-1511,\n-1512,\n-1513,\n-1514,\n-1515,\n-1516,\n-1517,\n-1518,\n-1519,\n-1520,\n-1521,\n-1522,\n-1523,\n-1524,\n-1525,\n-1526,\n-1527,\n-1528,\n-1529,\n-1530,\n-1531,\n-1532,\n-1533,\n-1534,\n-1535,\n-1536,\n-1537,\n-1538,\n-1539,\n-1540,\n-1541,\n-1542,\n-1543,\n-1544,\n-1545,\n-1546,\n-1547,\n-1548,\n-1549,\n-1550,\n-1551,\n-1552,\n-1553,\n-1554,\n-1555,\n-1556,\n-1557,\n-1558,\n-1559,\n-1560,\n-1561,\n-1562,\n-1563,\n-1564,\n-1565,\n-1566,\n-1567,\n-1568,\n-1569,\n-1570,\n-1571,\n-1572,\n-1573,\n-1574,\n-1575,\n-1576,\n-1577,\n-1578,\n-1579,\n-1580,\n-1581,\n-1582,\n-1583,\n-1584,\n-1585,\n-1586,\n-1587,\n-1588,\n-1589,\n-1590,\n-1591,\n-1592,\n-1593,\n-1594,\n-1595,\n-1596,\n-1597,\n-1598,\n-1599,\n-1600,\n-1601,\n-1602,\n-1603,\n-1604,\n-1605,\n-1606,\n-1607,\n-1608,\n-1609,\n-1610,\n-1611,\n-1612,\n-1613,\n-1614,\n-1615,\n-1616,\n-1617,\n-1618,\n-1619,\n-1620,\n-1621,\n-1622,\n-1623,\n-1624,\n-1625,\n-1626,\n-1627,\n-1628,\n-1629,\n-1630,\n-1631,\n-1632,\n-1633,\n-1634,\n-1635,\n-1636,\n-1637,\n-1638,\n-1639,\n-1640,\n-1641,\n-1642,\n-1643,\n-1644,\n-1645,\n-1646,\n-1647,\n-1648,\n-1649,\n-1650,\n-1651,\n-1652,\n-1653,\n-1654,\n-1655,\n-1656,\n-1657,\n-1658,\n-1659,\n-1660,\n-1661,\n-1662,\n-1663,\n-1664,\n-1665,\n-1666,\n-1667,\n-1668,\n-1669,\n-1670,\n-1671,\n-1672,\n-1673,\n-1674,\n-1675,\n-1676,\n-1677,\n-1678,\n-1679,\n-1680,\n-1681,\n-1682,\n-1683,\n-1684,\n-1685,\n-1686,\n-1687,\n-1688,\n-1689,\n-1690,\n-1691,\n-1692,\n-1693,\n-1694,\n-1695,\n-1696,\n-1697,\n-1698,\n-1699,\n-1700,\n-1701,\n-1702,\n-1703,\n-1704,\n-1705,\n-1706,\n-1707,\n-1708,\n-1709,\n-1710,\n-1711,\n-1712,\n-1713,\n-1714,\n-1715,\n-1716,\n-1717,\n-1718,\n-1719,\n-1720,\n-1721,\n-1722,\n-1809,\n-1810,\n+3,\n+9,\n+17,\n+23,\n+286,\n+287,\n+288,\n+482,\n+483,\n+484,\n+678,\n+940,\n+941,\n+942,\n+1136,\n+1137,\n+1138,\n+1332,\n+1811,\n+1812,\n+1813,\n+1814,\n+1815,\n+1816,\n+1817,\n+1818,\n+1819,\n+1820,\n+1821,\n+2736,\n+2737,\n+2738,\n+2739,\n+2740,\n+2741,\n+2742,\n+2743,\n+2744,\n+2745,\n+2746,\n' is not false : ccxtools write_inp_file test failed.
Comparing /home/mario/Downloads/FreeCAD-test/Mod/Fem/femtest/data/calculix/constraint_centrif.inp to /tmp/FEM_unittests/ccxtools_constraint_centrif_8ffda9a3155f/Mesh.inp failed!
@@ -18683,769 +18683,40 @@
-1,
-2,
-10,
-11,
-12,
-13,
-14,
-29,
-30,
-31,
-32,
-33,
-34,
-35,
-36,
-37,
-38,
-39,
-40,
-41,
-42,
-43,
-44,
-45,
-46,
-47,
-48,
-49,
-50,
-51,
-52,
-53,
-54,
-55,
-56,
-57,
-58,
-59,
-60,
-61,
-62,
-63,
-64,
-65,
-66,
-67,
-68,
-69,
-70,
-71,
-72,
-73,
-74,
-75,
-76,
-77,
-78,
-79,
-80,
-81,
-82,
-83,
-84,
-85,
-86,
-87,
-88,
-89,
-90,
-91,
-92,
-93,
-94,
-95,
-96,
-97,
-98,
-99,
-100,
-101,
-102,
-103,
-104,
-105,
-106,
-107,
-108,
-109,
-110,
-111,
-112,
-113,
-114,
-115,
-116,
-117,
-118,
-119,
-120,
-121,
-122,
-123,
-124,
-125,
-126,
-127,
-128,
-129,
-130,
-131,
-132,
-133,
-134,
-135,
-136,
-137,
-138,
-139,
-140,
-141,
-142,
-143,
-144,
-145,
-146,
-147,
-148,
-149,
-150,
-151,
-152,
-153,
-154,
-155,
-156,
-157,
-158,
-159,
-160,
-161,
-162,
-163,
-164,
-165,
-166,
-167,
-168,
-169,
-170,
-171,
-172,
-173,
-174,
-175,
-176,
-177,
-178,
-179,
-180,
-181,
-182,
-183,
-184,
-185,
-186,
-187,
-188,
-189,
-190,
-191,
-192,
-193,
-194,
-195,
-196,
-197,
-198,
-199,
-200,
-201,
-202,
-203,
-204,
-205,
-206,
-207,
-208,
-209,
-210,
-211,
-212,
-213,
-214,
-215,
-216,
-217,
-218,
-219,
-220,
-221,
-222,
-223,
-224,
-225,
-226,
-227,
-228,
-229,
-230,
-231,
-232,
-233,
-234,
-235,
-236,
-237,
-238,
-239,
-240,
-241,
-242,
-243,
-244,
-245,
-246,
-247,
-248,
-249,
-250,
-251,
-252,
-253,
-254,
-255,
-256,
-257,
-258,
-259,
-260,
-261,
-262,
-263,
-264,
-265,
-266,
-267,
-268,
-269,
-270,
-271,
-272,
-273,
-274,
-275,
-276,
-277,
-278,
-279,
-280,
-281,
-282,
-283,
-284,
-285,
-523,
-524,
-525,
-526,
-527,
-528,
-529,
-530,
-531,
-532,
-533,
-534,
-535,
-536,
-537,
-538,
-539,
-540,
-541,
-542,
-543,
-544,
-545,
-546,
-547,
-548,
-549,
-550,
-551,
-552,
-553,
-554,
-555,
-556,
-557,
-558,
-559,
-560,
-561,
-562,
-563,
-564,
-565,
-566,
-567,
-568,
-569,
-570,
-571,
-572,
-573,
-574,
-575,
-576,
-577,
-578,
-579,
-580,
-581,
-582,
-583,
-584,
-585,
-586,
-587,
-588,
-589,
-590,
-591,
-592,
-593,
-594,
-595,
-596,
-597,
-598,
-599,
-600,
-601,
-602,
-603,
-604,
-605,
-606,
-607,
-608,
-609,
-610,
-649,
-650,
-651,
-652,
-653,
-654,
-655,
-656,
-657,
-658,
-659,
-660,
-661,
-662,
-663,
-664,
-665,
-666,
-667,
-668,
-669,
-670,
-671,
-672,
-673,
-674,
-675,
-676,
-677,
-1337,
-1338,
-1339,
-1340,
-1341,
-1342,
-1343,
-1344,
-1345,
-1346,
-1347,
-1348,
-1349,
-1350,
-1351,
-1352,
-1353,
-1354,
-1355,
-1356,
-1357,
-1358,
-1359,
-1360,
-1361,
-1362,
-1363,
-1364,
-1365,
-1366,
-1367,
-1368,
-1369,
-1370,
-1371,
-1372,
-1373,
-1374,
-1375,
-1376,
-1377,
-1378,
-1379,
-1380,
-1381,
-1382,
-1383,
-1384,
-1385,
-1386,
-1387,
-1388,
-1389,
-1390,
-1391,
-1392,
-1393,
-1394,
-1395,
-1396,
-1397,
-1398,
-1399,
-1400,
-1401,
-1402,
-1403,
-1404,
-1405,
-1406,
-1407,
-1408,
-1409,
-1410,
-1411,
-1412,
-1413,
-1414,
-1415,
-1416,
-1417,
-1418,
-1419,
-1420,
-1421,
-1422,
-1423,
-1424,
-1425,
-1426,
-1427,
-1428,
-1429,
-1430,
-1431,
-1432,
-1433,
-1434,
-1435,
-1436,
-1437,
-1438,
-1439,
-1440,
-1441,
-1442,
-1443,
-1444,
-1445,
-1446,
-1447,
-1448,
-1449,
-1450,
-1451,
-1452,
-1453,
-1454,
-1455,
-1456,
-1457,
-1458,
-1459,
-1460,
-1461,
-1462,
-1463,
-1464,
-1465,
-1466,
-1467,
-1468,
-1469,
-1470,
-1471,
-1472,
-1473,
-1474,
-1475,
-1476,
-1477,
-1478,
-1479,
-1480,
-1481,
-1482,
-1483,
-1484,
-1485,
-1486,
-1487,
-1488,
-1489,
-1490,
-1491,
-1492,
-1493,
-1494,
-1495,
-1496,
-1497,
-1498,
-1499,
-1500,
-1501,
-1502,
-1503,
-1504,
-1505,
-1506,
-1507,
-1508,
-1509,
-1510,
-1511,
-1512,
-1513,
-1514,
-1515,
-1516,
-1517,
-1518,
-1519,
-1520,
-1521,
-1522,
-1523,
-1524,
-1525,
-1526,
-1527,
-1528,
-1529,
-1530,
-1531,
-1532,
-1533,
-1534,
-1535,
-1536,
-1537,
-1538,
-1539,
-1540,
-1541,
-1542,
-1543,
-1544,
-1545,
-1546,
-1547,
-1548,
-1549,
-1550,
-1551,
-1552,
-1553,
-1554,
-1555,
-1556,
-1557,
-1558,
-1559,
-1560,
-1561,
-1562,
-1563,
-1564,
-1565,
-1566,
-1567,
-1568,
-1569,
-1570,
-1571,
-1572,
-1573,
-1574,
-1575,
-1576,
-1577,
-1578,
-1579,
-1580,
-1581,
-1582,
-1583,
-1584,
-1585,
-1586,
-1587,
-1588,
-1589,
-1590,
-1591,
-1592,
-1593,
-1594,
-1595,
-1596,
-1597,
-1598,
-1599,
-1600,
-1601,
-1602,
-1603,
-1604,
-1605,
-1606,
-1607,
-1608,
-1609,
-1610,
-1611,
-1612,
-1613,
-1614,
-1615,
-1616,
-1617,
-1618,
-1619,
-1620,
-1621,
-1622,
-1623,
-1624,
-1625,
-1626,
-1627,
-1628,
-1629,
-1630,
-1631,
-1632,
-1633,
-1634,
-1635,
-1636,
-1637,
-1638,
-1639,
-1640,
-1641,
-1642,
-1643,
-1644,
-1645,
-1646,
-1647,
-1648,
-1649,
-1650,
-1651,
-1652,
-1653,
-1654,
-1655,
-1656,
-1657,
-1658,
-1659,
-1660,
-1661,
-1662,
-1663,
-1664,
-1665,
-1666,
-1667,
-1668,
-1669,
-1670,
-1671,
-1672,
-1673,
-1674,
-1675,
-1676,
-1677,
-1678,
-1679,
-1680,
-1681,
-1682,
-1683,
-1684,
-1685,
-1686,
-1687,
-1688,
-1689,
-1690,
-1691,
-1692,
-1693,
-1694,
-1695,
-1696,
-1697,
-1698,
-1699,
-1700,
-1701,
-1702,
-1703,
-1704,
-1705,
-1706,
-1707,
-1708,
-1709,
-1710,
-1711,
-1712,
-1713,
-1714,
-1715,
-1716,
-1717,
-1718,
-1719,
-1720,
-1721,
-1722,
-1809,
-1810,
+3,
+9,
+17,
+23,
+286,
+287,
+288,
+482,
+483,
+484,
+678,
+940,
+941,
+942,
+1136,
+1137,
+1138,
+1332,
+1811,
+1812,
+1813,
+1814,
+1815,
+1816,
+1817,
+1818,
+1819,
+1820,
+1821,
+2736,
+2737,
+2738,
+2739,
+2740,
+2741,
+2742,
+2743,
+2744,
+2745,
+2746,

```


test_constraint_contact-solid_solid:

```
Traceback (most recent call last):
  File "/usr/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/usr/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "FreeCAD-test/Mod/Fem/femtest/app/test_ccxtools.py", line 240, in test_constraint_contact_solid_solid
    self.input_file_writing_test(get_namefromdef("test_"))
  File "FreeCAD-test/Mod/Fem/femtest/app/test_ccxtools.py", line 372, in input_file_writing_test
    self.assertFalse(ret, f"ccxtools write_inp_file test failed.\n{ret}")
AssertionError: 'Comparing FreeCAD-test/Mod/Fem/femtest/data/calculix/constraint_contact_solid_solid.inp to /tmp/FEM_unittests/ccxtools_constraint_contact_solid_solid_d4764ec9353f/Mesh.inp failed!\n@@ -4348,0 +4349,5 @@\n+93,\n+94,\n+95,\n+96,\n+97,\n@@ -4351,155 +4356,7 @@\n-100,\n-101,\n-102,\n-103,\n-104,\n-105,\n-106,\n-107,\n-108,\n-109,\n-110,\n-111,\n-112,\n-113,\n-114,\n-115,\n-116,\n-117,\n-118,\n-119,\n-120,\n-121,\n-122,\n-123,\n-124,\n-125,\n-126,\n-127,\n-128,\n-129,\n-130,\n-131,\n-132,\n-133,\n-134,\n-135,\n-136,\n-137,\n-138,\n-139,\n-140,\n-141,\n-142,\n-143,\n-144,\n-145,\n-146,\n-147,\n-148,\n-149,\n-150,\n-151,\n-152,\n-153,\n-154,\n-155,\n-156,\n-157,\n-158,\n-159,\n-160,\n-161,\n-162,\n-163,\n-275,\n-276,\n-277,\n-278,\n-279,\n-280,\n-281,\n-282,\n-283,\n-284,\n-285,\n-286,\n-287,\n-288,\n-289,\n-290,\n-291,\n-292,\n-293,\n-294,\n-295,\n-296,\n-297,\n-298,\n-299,\n-300,\n-301,\n-302,\n-303,\n-304,\n-305,\n-306,\n-307,\n-308,\n-309,\n-310,\n-311,\n-312,\n-313,\n-314,\n-315,\n-316,\n-317,\n-318,\n-319,\n-320,\n-321,\n-322,\n-323,\n-324,\n-325,\n-326,\n-327,\n-328,\n-329,\n-330,\n-331,\n-332,\n-333,\n-334,\n-335,\n-336,\n-337,\n-338,\n-339,\n-340,\n-341,\n-342,\n-343,\n-344,\n-345,\n-346,\n-347,\n-348,\n-349,\n-350,\n-351,\n-352,\n-353,\n-354,\n-355,\n-356,\n-357,\n-358,\n-359,\n-360,\n-361,\n-362,\n-363,\n-364,\n-365,\n+164,\n+165,\n+166,\n+167,\n+168,\n+274,\n+366,\n@@ -4526,0 +4384,6 @@\n+652,\n+653,\n+654,\n+655,\n+656,\n+657,\n@@ -4530,66 +4393,6 @@\n-661,\n-662,\n-663,\n-664,\n-665,\n-666,\n-667,\n-668,\n-669,\n-670,\n-671,\n-672,\n-673,\n-674,\n-675,\n-676,\n-677,\n-678,\n-679,\n-680,\n-681,\n-682,\n-683,\n-684,\n-685,\n-686,\n-687,\n-688,\n-689,\n-690,\n-691,\n-692,\n-693,\n-694,\n-695,\n-696,\n-697,\n-698,\n-699,\n-700,\n-701,\n-702,\n-703,\n-704,\n-705,\n-706,\n-707,\n-708,\n-709,\n-710,\n-711,\n-712,\n-713,\n-714,\n-715,\n-716,\n-717,\n-718,\n-719,\n-720,\n-721,\n-722,\n-723,\n-724,\n-725,\n-726,\n+727,\n+728,\n+729,\n+730,\n+731,\n+732,\n@@ -4608,342 +4411,18 @@\n-1219,\n-1220,\n-1221,\n-1222,\n-1223,\n-1224,\n-1225,\n-1226,\n-1227,\n-1228,\n-1229,\n-1230,\n-1231,\n-1232,\n-1233,\n-1234,\n-1235,\n-1236,\n-1237,\n-1238,\n-1239,\n-1240,\n-1241,\n-1242,\n-1243,\n-1244,\n-1245,\n-1246,\n-1247,\n-1248,\n-1249,\n-1250,\n-1251,\n-1252,\n-1253,\n-1254,\n-1255,\n-1256,\n-1257,\n-1258,\n-1259,\n-1260,\n-1261,\n-1262,\n-1263,\n-1264,\n-1265,\n-1266,\n-1267,\n-1268,\n-1269,\n-1270,\n-1271,\n-1272,\n-1273,\n-1274,\n-1275,\n-1276,\n-1277,\n-1278,\n-1279,\n-1280,\n-1281,\n-1282,\n-1283,\n-1284,\n-1285,\n-1286,\n-1287,\n-1288,\n-1289,\n-1290,\n-1291,\n-1292,\n-1293,\n-1294,\n-1295,\n-1296,\n-1297,\n-1298,\n-1299,\n-1300,\n-1301,\n-1302,\n-1303,\n-1304,\n-1305,\n-1306,\n-1307,\n-1308,\n-1309,\n-1310,\n-1311,\n-1312,\n-1313,\n-1314,\n-1315,\n-1316,\n-1317,\n-1318,\n-1319,\n-1320,\n-1321,\n-1322,\n-1323,\n-1324,\n-1325,\n-1326,\n-1327,\n-1328,\n-1329,\n-1330,\n-1331,\n-1332,\n-1333,\n-1334,\n-1335,\n-1336,\n-1337,\n-1338,\n-1339,\n-1340,\n-1341,\n-1342,\n-1343,\n-1344,\n-1345,\n-1346,\n-1347,\n-1348,\n-1349,\n-1350,\n-1351,\n-1352,\n-1353,\n-1354,\n-1355,\n-1356,\n-1357,\n-1358,\n-1359,\n-1360,\n-1361,\n-1362,\n-1363,\n-1364,\n-1365,\n-1366,\n-1367,\n-1368,\n-1369,\n-1370,\n-1371,\n-1372,\n-1373,\n-1374,\n-1375,\n-1376,\n-1377,\n-1378,\n-1379,\n-1380,\n-1381,\n-1382,\n-1383,\n-1384,\n-1385,\n-1386,\n-1387,\n-1388,\n-1389,\n-1390,\n-1391,\n-1392,\n-1393,\n-1394,\n-1395,\n-1396,\n-1397,\n-1398,\n-1399,\n-1400,\n-1401,\n-1402,\n-1403,\n-1404,\n-1405,\n-1406,\n-1407,\n-1408,\n-1409,\n-1410,\n-1411,\n-1412,\n-1413,\n-1414,\n-1415,\n-1416,\n-1417,\n-1418,\n-1419,\n-1420,\n-1421,\n-1422,\n-1423,\n-1424,\n-1425,\n-1426,\n-1427,\n-1428,\n-1429,\n-1430,\n-1431,\n-1432,\n-1433,\n-1434,\n-1435,\n-1436,\n-1437,\n-1438,\n-1439,\n-1440,\n-1441,\n-1442,\n-1443,\n-1444,\n-1445,\n-1446,\n-1447,\n-1448,\n-1449,\n-1450,\n-1451,\n-1452,\n-1453,\n-1454,\n-1455,\n-1456,\n-1457,\n-1458,\n-1459,\n-1460,\n-1461,\n-1462,\n-1463,\n-1464,\n-1465,\n-1466,\n-1467,\n-1468,\n-1469,\n-1470,\n-1471,\n-1472,\n-1473,\n-1474,\n-1475,\n-1476,\n-1477,\n-1478,\n-1479,\n-1480,\n-1481,\n-1482,\n-1483,\n-1484,\n-1485,\n-1486,\n-1487,\n-1488,\n-1489,\n-1490,\n-1491,\n-1492,\n-1493,\n-1494,\n-1495,\n-1496,\n-1497,\n-1498,\n-1499,\n-1500,\n-1501,\n-1502,\n-1503,\n-1504,\n-1505,\n-1506,\n-1507,\n-1508,\n-1509,\n-1510,\n-1511,\n-1512,\n-1513,\n-1514,\n-1515,\n-1516,\n-1517,\n-1518,\n-1519,\n-1520,\n-1521,\n-1522,\n-1523,\n-1524,\n-1525,\n-1526,\n-1527,\n-1528,\n-1529,\n-1530,\n-1531,\n-1532,\n-1533,\n-1534,\n-1535,\n-1536,\n-1537,\n-1538,\n-1539,\n-1540,\n-1541,\n-1542,\n-1543,\n-1544,\n-1545,\n-1546,\n-1547,\n-1548,\n-1549,\n-1550,\n-1551,\n-1552,\n-1553,\n-1554,\n-1555,\n-1556,\n-1557,\n-1558,\n-1559,\n-1560,\n+1210,\n+1211,\n+1212,\n+1213,\n+1214,\n+1215,\n+1216,\n+1217,\n+1218,\n+1561,\n+1562,\n+1563,\n+1564,\n+1565,\n+1566,\n+1567,\n+1568,\n+1569,\n@@ -5604,9 +5083,252 @@\n-1733,P4,100.0\n-1793,P1,100.0\n-1981,P4,100.0\n-2040,P1,100.0\n-2304,P2,100.0\n-2380,P3,100.0\n-2463,P2,100.0\n-2470,P2,100.0\n-2547,P1,100.0\n+1322,P2,100.0\n+1516,P1,100.0\n+1701,P1,100.0\n+1727,P1,100.0\n+1728,P1,100.0\n+1729,P1,100.0\n+1733,P3,100.0\n+1745,P2,100.0\n+1746,P3,100.0\n+1751,P3,100.0\n+1754,P2,100.0\n+1756,P1,100.0\n+1757,P4,100.0\n+1759,P4,100.0\n+1761,P1,100.0\n+1762,P1,100.0\n+1763,P2,100.0\n+1765,P1,100.0\n+1773,P2,100.0\n+1775,P1,100.0\n+1778,P2,100.0\n+1779,P2,100.0\n+1781,P1,100.0\n+1782,P4,100.0\n+1786,P3,100.0\n+1803,P2,100.0\n+1804,P2,100.0\n+1805,P2,100.0\n+1806,P4,100.0\n+1807,P2,100.0\n+1809,P3,100.0\n+1810,P4,100.0\n+1812,P4,100.0\n+1813,P3,100.0\n+1815,P1,100.0\n+1817,P2,100.0\n+1821,P4,100.0\n+1824,P3,100.0\n+1825,P3,100.0\n+1826,P2,100.0\n+1828,P4,100.0\n+1830,P2,100.0\n+1834,P1,100.0\n+1836,P2,100.0\n+1839,P1,100.0\n+1840,P3,100.0\n+1842,P1,100.0\n+1846,P2,100.0\n+1847,P4,100.0\n+1848,P2,100.0\n+1852,P4,100.0\n+1854,P1,100.0\n+1866,P4,100.0\n+1869,P4,100.0\n+1875,P3,100.0\n+1881,P2,100.0\n+1882,P1,100.0\n+1883,P2,100.0\n+1885,P3,100.0\n+1886,P3,100.0\n+1888,P3,100.0\n+1895,P1,100.0\n+1896,P2,100.0\n+1897,P2,100.0\n+1902,P2,100.0\n+1905,P1,100.0\n+1914,P2,100.0\n+1921,P4,100.0\n+1933,P2,100.0\n+1935,P2,100.0\n+1940,P3,100.0\n+1941,P2,100.0\n+1946,P1,100.0\n+1947,P3,100.0\n+1953,P1,100.0\n+1958,P3,100.0\n+1961,P3,100.0\n+1966,P2,100.0\n+1976,P3,100.0\n+1983,P2,100.0\n+1989,P2,100.0\n+1993,P1,100.0\n+1994,P4,100.0\n+1996,P4,100.0\n+1998,P2,100.0\n+2001,P2,100.0\n+2008,P1,100.0\n+2010,P3,100.0\n+2018,P4,100.0\n+2024,P1,100.0\n+2030,P4,100.0\n+2031,P4,100.0\n+2042,P3,100.0\n+2044,P1,100.0\n+2046,P4,100.0\n+2047,P2,100.0\n+2056,P4,100.0\n+2057,P2,100.0\n+2071,P1,100.0\n+2075,P1,100.0\n+2076,P4,100.0\n+2079,P3,100.0\n+2080,P1,100.0\n+2081,P1,100.0\n+2082,P2,100.0\n+2083,P3,100.0\n+2088,P1,100.0\n+2089,P2,100.0\n+2090,P2,100.0\n+2091,P2,100.0\n+2093,P2,100.0\n+2107,P3,100.0\n+2108,P2,100.0\n+2116,P4,100.0\n+2119,P4,100.0\n+2122,P1,100.0\n+2125,P3,100.0\n+2128,P4,100.0\n+2135,P1,100.0\n+2136,P3,100.0\n+2140,P3,100.0\n+2141,P4,100.0\n+2146,P1,100.0\n+2147,P2,100.0\n+2151,P3,100.0\n+2152,P4,100.0\n+2154,P4,100.0\n+2155,P3,100.0\n+2156,P3,100.0\n+2160,P3,100.0\n+2164,P3,100.0\n+2166,P4,100.0\n+2167,P1,100.0\n+2168,P2,100.0\n+2169,P3,100.0\n+2171,P2,100.0\n+2172,P1,100.0\n+2174,P4,100.0\n+2176,P2,100.0\n+2179,P4,100.0\n+2183,P1,100.0\n+2189,P3,100.0\n+2191,P1,100.0\n+2192,P4,100.0\n+2197,P3,100.0\n+2198,P3,100.0\n+2200,P1,100.0\n+2204,P2,100.0\n+2206,P2,100.0\n+2211,P3,100.0\n+2216,P2,100.0\n+2218,P3,100.0\n+2220,P2,100.0\n+2222,P4,100.0\n+2228,P4,100.0\n+2229,P1,100.0\n+2231,P1,100.0\n+2235,P2,100.0\n+2238,P4,100.0\n+2241,P2,100.0\n+2246,P1,100.0\n+2247,P2,100.0\n+2250,P1,100.0\n+2251,P2,100.0\n+2259,P2,100.0\n+2262,P2,100.0\n+2264,P3,100.0\n+2266,P4,100.0\n+2268,P4,100.0\n+2269,P2,100.0\n+2272,P4,100.0\n+2273,P4,100.0\n+2276,P3,100.0\n+2278,P4,100.0\n+2279,P1,100.0\n+2286,P1,100.0\n+2290,P3,100.0\n+2292,P3,100.0\n+2296,P3,100.0\n+2307,P3,100.0\n+2310,P4,100.0\n+2312,P3,100.0\n+2314,P1,100.0\n+2318,P1,100.0\n+2321,P3,100.0\n+2323,P2,100.0\n+2325,P4,100.0\n+2327,P1,100.0\n+2336,P4,100.0\n+2344,P2,100.0\n+2346,P4,100.0\n+2347,P3,100.0\n+2349,P2,100.0\n+2355,P2,100.0\n+2358,P3,100.0\n+2361,P2,100.0\n+2363,P4,100.0\n+2369,P3,100.0\n+2372,P2,100.0\n+2380,P4,100.0\n+2390,P1,100.0\n+2392,P4,100.0\n+2396,P4,100.0\n+2397,P4,100.0\n+2400,P3,100.0\n+2404,P3,100.0\n+2406,P2,100.0\n+2408,P3,100.0\n+2425,P1,100.0\n+2430,P2,100.0\n+2436,P2,100.0\n+2439,P2,100.0\n+2442,P3,100.0\n+2451,P1,100.0\n+2452,P3,100.0\n+2462,P1,100.0\n+2464,P1,100.0\n+2472,P1,100.0\n+2477,P4,100.0\n+2478,P1,100.0\n+2482,P3,100.0\n+2488,P3,100.0\n+2491,P1,100.0\n+2493,P1,100.0\n+2496,P2,100.0\n+2499,P1,100.0\n+2505,P3,100.0\n+2512,P2,100.0\n+2514,P3,100.0\n+2521,P1,100.0\n+2526,P4,100.0\n+2530,P1,100.0\n+2534,P2,100.0\n+2536,P3,100.0\n+2537,P4,100.0\n+2542,P1,100.0\n+2546,P2,100.0\n+2548,P4,100.0\n+2550,P3,100.0\n+2552,P4,100.0\n+2556,P3,100.0\n+2557,P2,100.0\n+2559,P2,100.0\n+2560,P2,100.0\n+2570,P1,100.0\n+2580,P2,100.0\n+2584,P2,100.0\n+2585,P1,100.0\n+2586,P1,100.0\n+2589,P1,100.0\n+2592,P3,100.0\n+2596,P2,100.0\n' is not false : ccxtools write_inp_file test failed.
Comparing FreeCAD-test/Mod/Fem/femtest/data/calculix/constraint_contact_solid_solid.inp to /tmp/FEM_unittests/ccxtools_constraint_contact_solid_solid_d4764ec9353f/Mesh.inp failed!
@@ -4348,0 +4349,5 @@
+93,
+94,
+95,
+96,
+97,
@@ -4351,155 +4356,7 @@
-100,
-101,
-102,
-103,
-104,
-105,
-106,
-107,
-108,
-109,
-110,
-111,
-112,
-113,
-114,
-115,
-116,
-117,
-118,
-119,
-120,
-121,
-122,
-123,
-124,
-125,
-126,
-127,
-128,
-129,
-130,
-131,
-132,
-133,
-134,
-135,
-136,
-137,
-138,
-139,
-140,
-141,
-142,
-143,
-144,
-145,
-146,
-147,
-148,
-149,
-150,
-151,
-152,
-153,
-154,
-155,
-156,
-157,
-158,
-159,
-160,
-161,
-162,
-163,
-275,
-276,
-277,
-278,
-279,
-280,
-281,
-282,
-283,
-284,
-285,
-286,
-287,
-288,
-289,
-290,
-291,
-292,
-293,
-294,
-295,
-296,
-297,
-298,
-299,
-300,
-301,
-302,
-303,
-304,
-305,
-306,
-307,
-308,
-309,
-310,
-311,
-312,
-313,
-314,
-315,
-316,
-317,
-318,
-319,
-320,
-321,
-322,
-323,
-324,
-325,
-326,
-327,
-328,
-329,
-330,
-331,
-332,
-333,
-334,
-335,
-336,
-337,
-338,
-339,
-340,
-341,
-342,
-343,
-344,
-345,
-346,
-347,
-348,
-349,
-350,
-351,
-352,
-353,
-354,
-355,
-356,
-357,
-358,
-359,
-360,
-361,
-362,
-363,
-364,
-365,
+164,
+165,
+166,
+167,
+168,
+274,
+366,
@@ -4526,0 +4384,6 @@
+652,
+653,
+654,
+655,
+656,
+657,
@@ -4530,66 +4393,6 @@
-661,
-662,
-663,
-664,
-665,
-666,
-667,
-668,
-669,
-670,
-671,
-672,
-673,
-674,
-675,
-676,
-677,
-678,
-679,
-680,
-681,
-682,
-683,
-684,
-685,
-686,
-687,
-688,
-689,
-690,
-691,
-692,
-693,
-694,
-695,
-696,
-697,
-698,
-699,
-700,
-701,
-702,
-703,
-704,
-705,
-706,
-707,
-708,
-709,
-710,
-711,
-712,
-713,
-714,
-715,
-716,
-717,
-718,
-719,
-720,
-721,
-722,
-723,
-724,
-725,
-726,
+727,
+728,
+729,
+730,
+731,
+732,
@@ -4608,342 +4411,18 @@
-1219,
-1220,
-1221,
-1222,
-1223,
-1224,
-1225,
-1226,
-1227,
-1228,
-1229,
-1230,
-1231,
-1232,
-1233,
-1234,
-1235,
-1236,
-1237,
-1238,
-1239,
-1240,
-1241,
-1242,
-1243,
-1244,
-1245,
-1246,
-1247,
-1248,
-1249,
-1250,
-1251,
-1252,
-1253,
-1254,
-1255,
-1256,
-1257,
-1258,
-1259,
-1260,
-1261,
-1262,
-1263,
-1264,
-1265,
-1266,
-1267,
-1268,
-1269,
-1270,
-1271,
-1272,
-1273,
-1274,
-1275,
-1276,
-1277,
-1278,
-1279,
-1280,
-1281,
-1282,
-1283,
-1284,
-1285,
-1286,
-1287,
-1288,
-1289,
-1290,
-1291,
-1292,
-1293,
-1294,
-1295,
-1296,
-1297,
-1298,
-1299,
-1300,
-1301,
-1302,
-1303,
-1304,
-1305,
-1306,
-1307,
-1308,
-1309,
-1310,
-1311,
-1312,
-1313,
-1314,
-1315,
-1316,
-1317,
-1318,
-1319,
-1320,
-1321,
-1322,
-1323,
-1324,
-1325,
-1326,
-1327,
-1328,
-1329,
-1330,
-1331,
-1332,
-1333,
-1334,
-1335,
-1336,
-1337,
-1338,
-1339,
-1340,
-1341,
-1342,
-1343,
-1344,
-1345,
-1346,
-1347,
-1348,
-1349,
-1350,
-1351,
-1352,
-1353,
-1354,
-1355,
-1356,
-1357,
-1358,
-1359,
-1360,
-1361,
-1362,
-1363,
-1364,
-1365,
-1366,
-1367,
-1368,
-1369,
-1370,
-1371,
-1372,
-1373,
-1374,
-1375,
-1376,
-1377,
-1378,
-1379,
-1380,
-1381,
-1382,
-1383,
-1384,
-1385,
-1386,
-1387,
-1388,
-1389,
-1390,
-1391,
-1392,
-1393,
-1394,
-1395,
-1396,
-1397,
-1398,
-1399,
-1400,
-1401,
-1402,
-1403,
-1404,
-1405,
-1406,
-1407,
-1408,
-1409,
-1410,
-1411,
-1412,
-1413,
-1414,
-1415,
-1416,
-1417,
-1418,
-1419,
-1420,
-1421,
-1422,
-1423,
-1424,
-1425,
-1426,
-1427,
-1428,
-1429,
-1430,
-1431,
-1432,
-1433,
-1434,
-1435,
-1436,
-1437,
-1438,
-1439,
-1440,
-1441,
-1442,
-1443,
-1444,
-1445,
-1446,
-1447,
-1448,
-1449,
-1450,
-1451,
-1452,
-1453,
-1454,
-1455,
-1456,
-1457,
-1458,
-1459,
-1460,
-1461,
-1462,
-1463,
-1464,
-1465,
-1466,
-1467,
-1468,
-1469,
-1470,
-1471,
-1472,
-1473,
-1474,
-1475,
-1476,
-1477,
-1478,
-1479,
-1480,
-1481,
-1482,
-1483,
-1484,
-1485,
-1486,
-1487,
-1488,
-1489,
-1490,
-1491,
-1492,
-1493,
-1494,
-1495,
-1496,
-1497,
-1498,
-1499,
-1500,
-1501,
-1502,
-1503,
-1504,
-1505,
-1506,
-1507,
-1508,
-1509,
-1510,
-1511,
-1512,
-1513,
-1514,
-1515,
-1516,
-1517,
-1518,
-1519,
-1520,
-1521,
-1522,
-1523,
-1524,
-1525,
-1526,
-1527,
-1528,
-1529,
-1530,
-1531,
-1532,
-1533,
-1534,
-1535,
-1536,
-1537,
-1538,
-1539,
-1540,
-1541,
-1542,
-1543,
-1544,
-1545,
-1546,
-1547,
-1548,
-1549,
-1550,
-1551,
-1552,
-1553,
-1554,
-1555,
-1556,
-1557,
-1558,
-1559,
-1560,
+1210,
+1211,
+1212,
+1213,
+1214,
+1215,
+1216,
+1217,
+1218,
+1561,
+1562,
+1563,
+1564,
+1565,
+1566,
+1567,
+1568,
+1569,
@@ -5604,9 +5083,252 @@
-1733,P4,100.0
-1793,P1,100.0
-1981,P4,100.0
-2040,P1,100.0
-2304,P2,100.0
-2380,P3,100.0
-2463,P2,100.0
-2470,P2,100.0
-2547,P1,100.0
+1322,P2,100.0
+1516,P1,100.0
+1701,P1,100.0
+1727,P1,100.0
+1728,P1,100.0
+1729,P1,100.0
+1733,P3,100.0
+1745,P2,100.0
+1746,P3,100.0
+1751,P3,100.0
+1754,P2,100.0
+1756,P1,100.0
+1757,P4,100.0
+1759,P4,100.0
+1761,P1,100.0
+1762,P1,100.0
+1763,P2,100.0
+1765,P1,100.0
+1773,P2,100.0
+1775,P1,100.0
+1778,P2,100.0
+1779,P2,100.0
+1781,P1,100.0
+1782,P4,100.0
+1786,P3,100.0
+1803,P2,100.0
+1804,P2,100.0
+1805,P2,100.0
+1806,P4,100.0
+1807,P2,100.0
+1809,P3,100.0
+1810,P4,100.0
+1812,P4,100.0
+1813,P3,100.0
+1815,P1,100.0
+1817,P2,100.0
+1821,P4,100.0
+1824,P3,100.0
+1825,P3,100.0
+1826,P2,100.0
+1828,P4,100.0
+1830,P2,100.0
+1834,P1,100.0
+1836,P2,100.0
+1839,P1,100.0
+1840,P3,100.0
+1842,P1,100.0
+1846,P2,100.0
+1847,P4,100.0
+1848,P2,100.0
+1852,P4,100.0
+1854,P1,100.0
+1866,P4,100.0
+1869,P4,100.0
+1875,P3,100.0
+1881,P2,100.0
+1882,P1,100.0
+1883,P2,100.0
+1885,P3,100.0
+1886,P3,100.0
+1888,P3,100.0
+1895,P1,100.0
+1896,P2,100.0
+1897,P2,100.0
+1902,P2,100.0
+1905,P1,100.0
+1914,P2,100.0
+1921,P4,100.0
+1933,P2,100.0
+1935,P2,100.0
+1940,P3,100.0
+1941,P2,100.0
+1946,P1,100.0
+1947,P3,100.0
+1953,P1,100.0
+1958,P3,100.0
+1961,P3,100.0
+1966,P2,100.0
+1976,P3,100.0
+1983,P2,100.0
+1989,P2,100.0
+1993,P1,100.0
+1994,P4,100.0
+1996,P4,100.0
+1998,P2,100.0
+2001,P2,100.0
+2008,P1,100.0
+2010,P3,100.0
+2018,P4,100.0
+2024,P1,100.0
+2030,P4,100.0
+2031,P4,100.0
+2042,P3,100.0
+2044,P1,100.0
+2046,P4,100.0
+2047,P2,100.0
+2056,P4,100.0
+2057,P2,100.0
+2071,P1,100.0
+2075,P1,100.0
+2076,P4,100.0
+2079,P3,100.0
+2080,P1,100.0
+2081,P1,100.0
+2082,P2,100.0
+2083,P3,100.0
+2088,P1,100.0
+2089,P2,100.0
+2090,P2,100.0
+2091,P2,100.0
+2093,P2,100.0
+2107,P3,100.0
+2108,P2,100.0
+2116,P4,100.0
+2119,P4,100.0
+2122,P1,100.0
+2125,P3,100.0
+2128,P4,100.0
+2135,P1,100.0
+2136,P3,100.0
+2140,P3,100.0
+2141,P4,100.0
+2146,P1,100.0
+2147,P2,100.0
+2151,P3,100.0
+2152,P4,100.0
+2154,P4,100.0
+2155,P3,100.0
+2156,P3,100.0
+2160,P3,100.0
+2164,P3,100.0
+2166,P4,100.0
+2167,P1,100.0
+2168,P2,100.0
+2169,P3,100.0
+2171,P2,100.0
+2172,P1,100.0
+2174,P4,100.0
+2176,P2,100.0
+2179,P4,100.0
+2183,P1,100.0
+2189,P3,100.0
+2191,P1,100.0
+2192,P4,100.0
+2197,P3,100.0
+2198,P3,100.0
+2200,P1,100.0
+2204,P2,100.0
+2206,P2,100.0
+2211,P3,100.0
+2216,P2,100.0
+2218,P3,100.0
+2220,P2,100.0
+2222,P4,100.0
+2228,P4,100.0
+2229,P1,100.0
+2231,P1,100.0
+2235,P2,100.0
+2238,P4,100.0
+2241,P2,100.0
+2246,P1,100.0
+2247,P2,100.0
+2250,P1,100.0
+2251,P2,100.0
+2259,P2,100.0
+2262,P2,100.0
+2264,P3,100.0
+2266,P4,100.0
+2268,P4,100.0
+2269,P2,100.0
+2272,P4,100.0
+2273,P4,100.0
+2276,P3,100.0
+2278,P4,100.0
+2279,P1,100.0
+2286,P1,100.0
+2290,P3,100.0
+2292,P3,100.0
+2296,P3,100.0
+2307,P3,100.0
+2310,P4,100.0
+2312,P3,100.0
+2314,P1,100.0
+2318,P1,100.0
+2321,P3,100.0
+2323,P2,100.0
+2325,P4,100.0
+2327,P1,100.0
+2336,P4,100.0
+2344,P2,100.0
+2346,P4,100.0
+2347,P3,100.0
+2349,P2,100.0
+2355,P2,100.0
+2358,P3,100.0
+2361,P2,100.0
+2363,P4,100.0
+2369,P3,100.0
+2372,P2,100.0
+2380,P4,100.0
+2390,P1,100.0
+2392,P4,100.0
+2396,P4,100.0
+2397,P4,100.0
+2400,P3,100.0
+2404,P3,100.0
+2406,P2,100.0
+2408,P3,100.0
+2425,P1,100.0
+2430,P2,100.0
+2436,P2,100.0
+2439,P2,100.0
+2442,P3,100.0
+2451,P1,100.0
+2452,P3,100.0
+2462,P1,100.0
+2464,P1,100.0
+2472,P1,100.0
+2477,P4,100.0
+2478,P1,100.0
+2482,P3,100.0
+2488,P3,100.0
+2491,P1,100.0
+2493,P1,100.0
+2496,P2,100.0
+2499,P1,100.0
+2505,P3,100.0
+2512,P2,100.0
+2514,P3,100.0
+2521,P1,100.0
+2526,P4,100.0
+2530,P1,100.0
+2534,P2,100.0
+2536,P3,100.0
+2537,P4,100.0
+2542,P1,100.0
+2546,P2,100.0
+2548,P4,100.0
+2550,P3,100.0
+2552,P4,100.0
+2556,P3,100.0
+2557,P2,100.0
+2559,P2,100.0
+2560,P2,100.0
+2570,P1,100.0
+2580,P2,100.0
+2584,P2,100.0
+2585,P1,100.0
+2586,P1,100.0
+2589,P1,100.0
+2592,P3,100.0
+2596,P2,100.0
```

FEA-eng commented:
We agree with @marioalexis84 that it's a really large PR and it would be better to leave it for the next release cycle. @maxwxyz Can you mark it as Post 1.0?

lyphrowny commented:
@marioalexis84 It seems the issue with differences in .inp files runs deeper. For the `constaint_centrif.inp` case, I'm getting these values in `NSET=ConstraintFixed`:
```
1,
2,
10,
```
While you're getting these values:
```
3,
9,
17,
```

Why do I believe the issue is deeper? I ran the `constaint_centrif` test case and got my values. After that I updated `constraint_centrif.inp` with my values and passed the tests on GitHub. However, you're getting the `constaint_centrif.inp` with different values - the ones, that were before my changes. 
So, we're getting different values - does it depend on different OS, settings or something else? This needs further investigation and not as a part of this PR, I believe

I propose that we create an issue regarding these test cases and revert the changes made to them in this PR: 
- Revert the changes to the *.inp files
- Comment out these test cases (as they were)
 
What do you think?

marioalexis84 commented:
I guess you are generating the .inp files with the Part workbench preference parameter `Automatically refine model after boolean operation` unchecked. This pull request should fix the issue: https://github.com/FreeCAD/FreeCAD/pull/17204.
If you open both examples, you will see that the constraint assignments do not make sense. This is because the shapes in both examples were refined implicitly (whoever created the examples had the Part Workbench parameter `RefineModel = True`) before constraint assignments. Then if the examples are run on a system with another configuration the constraints are assigned to the wrong faces, so the .inp files do not match.

The error is generated because there is an improper dependency of the Part and PartDesign classes on configuration parameters.

@maxwxyz In this pull request for example https://github.com/FreeCAD/FreeCAD/pull/14406, Refine values ​​dependent on configuration parameters should be assigned in the  command activation functions where the objects are created, not inside the feature classes.
 
Here are some comments from wwmayer on this: https://github.com/FreeCAD/FreeCAD/issues/13383#issuecomment-2112217859

chennes commented:
@marioalexis84 what is the status here?

marioalexis84 commented:
The branch should be rebased without modifying the .inp of the centrif example.
But first this commit https://github.com/FreeCAD/FreeCAD/pull/18090 should be merged which fixes the references (boundary conditions are not assigned correctly).
On the other hand, I don't know how stable tests like centrif are, is there any guarantee that different versions of OCCT keep the same numbering of the faces in the shapes?

chennes commented:
@lyphrowny can you please take a look at @marioalexis84's comment above?

maxwxyz commented:
@lyphrowny waiting for feedback

maxwxyz commented:
@lyphrowny ping

maxwxyz commented:
@lyphrowny friendly reminder

luzpaz commented:
@lyphrowny ping

chennes committed:
Merge branch 'main' into fix_parse_diff

luzpaz commented:
What's the status of this PR @lyphrowny  ?
